src/components/ProductList/useFiltering.js

```diff
@@ -0,0 +1,29 @@
+import {useState, useEffect} from 'react'
+
+const useFiltering = (data) => {
+    const [filters, setFilters] = useState([]);
+    const [filteredData, setFilteredData] = useState(data);
+
+    useEffect(() => {
+        if(filters.length > 0){
```

Comment:
An easier way to achieve this, well not easier but more easy to read 

```suggestion
        filters.length > 0 ? setFilteredData(data.filter(x => filters.includes(x.data.category.slug))) : setFilteredData(data)
```